api/middlewares/auth.js

```diff
@@ -0,0 +1,31 @@
+const jwt = require("jsonwebtoken");
+const userDao = require("../models/userDao");
+
+const checkToken = async (req, res, next) => {
+  try {
+    const token = req.headers.authorization;
+
+    if (!token) {
+      const error = new Error("NOT_EXIST_TOKEN");
+      throw error;
+    }
+    const decoded = await jwt.verify(token, process.env.SECRET_KEY);
+
+    const user = await userDao.checkUserId(decoded.userId);
+
+    if (!user) {
+      const error = new Error("INVALID_TOKEN");
+      error.statusCode = 400;
+      throw error;
+    }
+    req.user = decoded.userId;
+    next();
+  } catch (error) {
+    error.statusCode = 400;
+    next(error);
+  }
```

Comment:
해당 미들웨어는 catchAsync로 감싸져 있지 않기 때문에 next로 error를 보내는 것이 아니고 return으로 응답을 보내야 합니다.
```suggestion
  } catch (error) {
    error.statusCode = 401;
    return res.status(error.statusCode).json({message: error.message});
  }
```